File: SpaceChallenge_Java/app/src/main/java/com/example/spacechallenge_java/Simulation.java
package com.example.spacechallenge_java;

import android.content.res.AssetManager;
import android.content.Context;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;

public class Simulation {
    private Context context;
    private String[] linesInFile;

    public Simulation(Context context) {
        this.context = context;
    }
    public ArrayList<Item> loadItems(String fileName) throws IOException {
        AssetManager asset = context.getAssets();
        ArrayList<Item> items = new ArrayList<>();
        Scanner scanedFiles = new Scanner(asset.open(fileName));
        while (scanedFiles.hasNextLine()) {
            linesInFile = scanedFiles.nextLine().split("=");
            Item item = new Item(linesInFile[0], Integer.parseInt(linesInFile[1]));
            items.add(item);
        }
        return items;
    }

    public ArrayList<Rocket> loadU1(ArrayList<Item> itemList) {
        ArrayList<Rocket> rocketU1List = new ArrayList();
        U1 rocketU1 = new U1();
        for (Item item : itemList) {
            if (rocketU1.canCarry(item)) rocketU1.carry(item);
            else {
                rocketU1List.add(rocketU1);
                rocketU1 = new U1();
                rocketU1.carry(item);
            }
        }
        return rocketU1List;
    }

    public ArrayList<Rocket> loadU2(ArrayList<Item> itemList) {
        ArrayList<Rocket> rocketU2List = new ArrayList();
        U2 rocketU2 = new U2();
        for (Item item : itemList) {
            if (rocketU2.canCarry(ite)) rocketU2.carry(item);
            else {
                rocketU2List.add(rocketU2);
                rocketU2 = new U2();
                rocketU2.carry(item);
            }
        }
        return rocketU2List;
    }

    public int runSimulation(ArrayList<Rocket> rocketList) {
        int budget = 0;
        for (Rocket rocket : rocketList) {
            budget += rocket.getCost();
            while (!rocket.land() || !rocket.launch()) {
                budget += rocket.getCost();
            }
        }
        return budget;
    }

}

File: SpaceChallenge_Kotlin/app/src/main/java/com/savelievoleksandr/spacechallenge_kotlin/Simulation.kt
package com.savelievoleksandr.spacechallenge_kotlin

import android.content.Context
import java.util.*
import kotlin.collections.ArrayList

class Simulation(private val context: Context) {
    fun loadItems(fileName: String): ArrayList<Item> {
        val asset = context.assets
        val items = ArrayList<Item>()
        val scanedFiles = Scanner(asset.open(fileName))
        while (scanedFiles.hasNextLine()) {
            val linesInFile = scanedFiles.nextLine().split("=".toRegex()).toTypedArray()
            val item = Item(linesInFile[0], linesInFile[1].toInt())
            items.add(item)
        }
        return items
    }

    fun loadU1(itemList: ArrayList<Item>): ArrayList<Rocket> {
        val rocketU1List: ArrayList<Rocket> = ArrayList()
        var rocketU1 = U1()
        for (item in itemList) {
            if (rocketU1.canCarry(item)) rocketU1.carry(item) else {
                rocketU1List.add(rocketU1)
                rocketU1 = U1()
                rocketU1.carry(item)
            }
        }
        return rocketU1List
    }

    fun loadU2(itemList: ArrayList<Item>): ArrayList<Rocket> {
        val rocketU2List: ArrayList<Rocket> = ArrayList()
        var rocketU2 = U2()
        for (item in itemList) {
            if (rocketU2.canCarry(item)) rocketU2.carry(item) else {
                rocketU2List.add(rocketU2)
                rocketU2 = U2()
                rocketU2.carry(item)
            }
        }
        return rocketU2List
    }

    fun runSimulation(rocketList: ArrayList<Rocket>): Int {
        var budget = 0
        for (rocket in rocketList) {
            budget += rocket.cost
            while (!rocket.land() || !rocket.launch()) {
                budget += rocket.cost
            }
        }
        return budget
    }
}
